posix/posix.c

```diff
@@ -214,19 +211,49 @@ int posix_getOpenFile(int fd, open_file_t **f)
 }
 
 
-int posix_newFile(process_info_t *p, int fd)
+static int _posix_allocfd(process_info_t *p, int fd)
 {
-	open_file_t *f;
-
-	proc_lockSet(&p->lock);
+	fildes_t *nfds;
+	int nfdsz;
 
 	for (; fd < p->maxfd; ++fd) {
+		if (fd >= p->fdsz) {
+			nfdsz = 2 * p->fdsz;
```

Comment:
when `p->fdsz == p->maxfd`, whenever user tries to alocate new file we reallocate memory, I'd add a check to prevent that

Reply:
For what value of `fd` will this occur?

Reply:
if `p->fdsz == p->maxfd` and `fd < p->maxfd` we'll never go into `fd >= p->fdsz` case, so no relocation will occur.
if `p->fdsz == p->maxfd` and `fd >= p->maxfd` we won't enter `for` loop and we'll return `-1`

Yes, we allow `fd == p->fdsz`, but please note that we deny `fd == p->maxfd` using `for` condition.

Or am I missing something here?